src/main/scala/org/broadinstitute/hail/utils/richUtils/RichHadoopConfiguration.scala

```diff
@@ -107,7 +110,7 @@ class RichHadoopConfiguration(val hConf: hadoop.conf.Configuration) extends AnyV
       false, hConf)
   }
 
-  def copyMerge(srcFilenames: Array[String], destFilename: String, deleteSource: Boolean = true) {
+  def copyMerge(srcFilenames: Array[(String, (FileStatus) => Int)], destFilename: String, deleteSource: Boolean = true) {
```

Comment:
I think it's better to have the caller to copyMerge say explicitly how to sort the files. If you absolutely hate it, I can try and think about how to incorporate the header separately in copyMerge.